hack/e2e-common.sh

```diff
@@ -450,6 +450,11 @@ function upgrade_test_flow {
     
     # Step 2: Create test resources
     echo "Creating test resources..."
+    
+    # Create custom namespace for test resources
+    kubectl create ns kueue-upgrade-test --kubeconfig="$1"
```

Comment:
I'm wondering how easy / hard it is to re-run the tests on a cluster? Is it basically one off, or we could support running in a loop? 

Do we have some teardown procedure for the resources?